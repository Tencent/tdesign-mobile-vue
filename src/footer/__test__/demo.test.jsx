/**
 * 该文件为由脚本 `npm run test:demo` 自动生成，如需修改，执行脚本命令即可。请勿手写直接修改，否则会被覆盖
 */

import { mount } from '@vue/test-utils';
import baseVue from '@/footer/demos/base.vue';
import linkVue from '@/footer/demos/link.vue';
import logoVue from '@/footer/demos/logo.vue';
import mobileVue from '@/footer/demos/mobile.vue';

const mapper = {
  baseVue,
  linkVue,
  logoVue,
  mobileVue,
};

describe('Footer', () => {
  Object.keys(mapper).forEach((demoName) => {
    it(`Footer ${demoName} demo works fine`, () => {
      const wrapper = mount(mapper[demoName]);
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
