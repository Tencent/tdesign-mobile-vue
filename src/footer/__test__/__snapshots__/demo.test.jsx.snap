// Vitest Snapshot v1

exports[`Footer > Footer baseVue demo works fine 1`] = `
<div
  class="t-footer"
>
  <!---->
  <!---->
  <div
    class="t-footer__text"
  >
    Copyright © 2019-2023 TDesign.All Rights Reserved.
  </div>
</div>
`;

exports[`Footer > Footer linkVue demo works fine 1`] = `
<div
  data-v-app=""
>
  
  <!-- theme 为 text，含有底部链接 -->
  <div
    class="footer-example"
  >
    <div
      class="t-footer"
    >
      <!---->
      <div
        class="t-footer__link-list"
      >
        
        
        <a
          class="t-footer__link-item"
          href=""
        >
          底部链接
        </a>
        <!---->
        
        
      </div>
      <div
        class="t-footer__text"
      >
        Copyright © 2021-2031 TD.All Rights Reserved.
      </div>
    </div>
  </div>
  <div
    class="footer-example"
  >
    <div
      class="t-footer"
    >
      <!---->
      <div
        class="t-footer__link-list"
      >
        
        
        <a
          class="t-footer__link-item"
          href=""
        >
          底部链接
        </a>
        <div
          class="t-footer__link-line"
        >
          |
        </div>
        
        
        <a
          class="t-footer__link-item"
          href=""
        >
          底部链接
        </a>
        <!---->
        
        
      </div>
      <div
        class="t-footer__text"
      >
        Copyright © 2021-2031 TD.All Rights Reserved.
      </div>
    </div>
  </div>
  
</div>
`;

exports[`Footer > Footer logoVue demo works fine 1`] = `
<div
  data-v-app=""
>
  
  <!-- theme 为 logo -->
  <div
    class="t-footer"
  >
    <a
      class="t-footer__logo"
    >
      <div
        class="t-image t-image--square t-footer__icon"
      >
        <div
          class="t-image__mask"
        >
          <div
            class="t-loading"
            style="font-size: 20px;"
          >
            <div
              class="t-loading__dots"
              style="animation-duration: 800ms; width: 20px; height: 20px;"
            >
              
              <div
                class="t-loading__dot"
                style="animation-duration: 0.8s; animation-delay: 0s;"
              />
              <div
                class="t-loading__dot"
                style="animation-duration: 0.8s; animation-delay: 0.26666666666666666s;"
              />
              <div
                class="t-loading__dot"
                style="animation-duration: 0.8s; animation-delay: 0.5333333333333333s;"
              />
              
            </div>
            <!---->
            <!---->
          </div>
        </div>
        <picture>
          <!--v-if-->
          <img
            alt=""
            class="t-image__img"
            src="https://tdesign.gtimg.com/mobile/demos/logo2.png"
            style="object-fit: fill; object-position: center;"
          />
        </picture>
      </div>
      <span
        class="t-footer__title"
      >
        品牌名称
      </span>
    </a>
    <!---->
    <!---->
  </div>
  
</div>
`;

exports[`Footer > Footer mobileVue demo works fine 1`] = `
<div
  class="tdesign-mobile-demo"
>
  <h1
    class="title"
  >
    Footer 页脚
  </h1>
  <p
    class="summary"
  >
    用于开启一个闭环的操作任务，如“删除”对象、“购买”商品等。
  </p>
  <div
    class="tdesign-mobile-demo-block tdesign-mobile-demo-block_notitle"
  >
    <div
      class="tdesign-mobile-demo-block__header"
    >
      <!--v-if-->
      <p
        class="tdesign-mobile-demo-block__summary tdesign-mobile-demo-block_subtitle"
      >
        基础页脚
      </p>
    </div>
    <div
      class="tdesign-mobile-demo-block__slot"
    >
      
      <div
        class="t-footer"
      >
        <!---->
        <!---->
        <div
          class="t-footer__text"
        >
          Copyright © 2019-2023 TDesign.All Rights Reserved.
        </div>
      </div>
      
    </div>
  </div>
  <div
    class="tdesign-mobile-demo-block tdesign-mobile-demo-block_notitle"
  >
    <div
      class="tdesign-mobile-demo-block__header"
    >
      <!--v-if-->
      <p
        class="tdesign-mobile-demo-block__summary tdesign-mobile-demo-block_subtitle"
      >
        基础加链接页脚
      </p>
    </div>
    <div
      class="tdesign-mobile-demo-block__slot"
    >
      
      
      <!-- theme 为 text，含有底部链接 -->
      <div
        class="footer-example"
      >
        <div
          class="t-footer"
        >
          <!---->
          <div
            class="t-footer__link-list"
          >
            
            
            <a
              class="t-footer__link-item"
              href=""
            >
              底部链接
            </a>
            <!---->
            
            
          </div>
          <div
            class="t-footer__text"
          >
            Copyright © 2021-2031 TD.All Rights Reserved.
          </div>
        </div>
      </div>
      <div
        class="footer-example"
      >
        <div
          class="t-footer"
        >
          <!---->
          <div
            class="t-footer__link-list"
          >
            
            
            <a
              class="t-footer__link-item"
              href=""
            >
              底部链接
            </a>
            <div
              class="t-footer__link-line"
            >
              |
            </div>
            
            
            <a
              class="t-footer__link-item"
              href=""
            >
              底部链接
            </a>
            <!---->
            
            
          </div>
          <div
            class="t-footer__text"
          >
            Copyright © 2021-2031 TD.All Rights Reserved.
          </div>
        </div>
      </div>
      
      
    </div>
  </div>
  <div
    class="tdesign-mobile-demo-block tdesign-mobile-demo-block_notitle"
  >
    <div
      class="tdesign-mobile-demo-block__header"
    >
      <!--v-if-->
      <p
        class="tdesign-mobile-demo-block__summary tdesign-mobile-demo-block_subtitle"
      >
        品牌页脚
      </p>
    </div>
    <div
      class="tdesign-mobile-demo-block__slot"
    >
      
      
      <!-- theme 为 logo -->
      <div
        class="t-footer"
      >
        <a
          class="t-footer__logo"
        >
          <div
            class="t-image t-image--square t-footer__icon"
          >
            <div
              class="t-image__mask"
            >
              <div
                class="t-loading"
                style="font-size: 20px;"
              >
                <div
                  class="t-loading__dots"
                  style="animation-duration: 800ms; width: 20px; height: 20px;"
                >
                  
                  <div
                    class="t-loading__dot"
                    style="animation-duration: 0.8s; animation-delay: 0s;"
                  />
                  <div
                    class="t-loading__dot"
                    style="animation-duration: 0.8s; animation-delay: 0.26666666666666666s;"
                  />
                  <div
                    class="t-loading__dot"
                    style="animation-duration: 0.8s; animation-delay: 0.5333333333333333s;"
                  />
                  
                </div>
                <!---->
                <!---->
              </div>
            </div>
            <picture>
              <!--v-if-->
              <img
                alt=""
                class="t-image__img"
                src="https://tdesign.gtimg.com/mobile/demos/logo2.png"
                style="object-fit: fill; object-position: center;"
              />
            </picture>
          </div>
          <span
            class="t-footer__title"
          >
            品牌名称
          </span>
        </a>
        <!---->
        <!---->
      </div>
      
      
    </div>
  </div>
</div>
`;
